OpenRA.Game/GameSpeed.cs

```diff
@@ -28,8 +28,27 @@ public class GameSpeeds : IGlobalModData
 		static object LoadSpeeds(MiniYaml y)
 		{
 			var ret = new Dictionary<string, GameSpeed>();
+			var defaultCount = 0;
 			foreach (var node in y.Nodes)
-				ret.Add(node.Key, FieldLoader.Load<GameSpeed>(node.Value));
+			{
+				var gs = FieldLoader.Load<GameSpeed>(node.Value);
+				if (string.IsNullOrEmpty(gs.Name))
+					throw new YamlException("At least one gamespeed doesn't define a Name!");
+				else if (gs.Timestep < 1)
+					throw new YamlException("Gamespeed {0} doesn't define a valid Timestep!".F(gs.Name));
+				else if (gs.OrderLatency < 1)
+					throw new YamlException("Gamespeed {0} doesn't define a valid OrderLatency!".F(gs.Name));
+
+				if (node.Key == "default")
+					++defaultCount;
+
+				ret.Add(node.Key, gs);
+			}
+
+			if (defaultCount == 0)
+				throw new YamlException("Mod doesn't define a 'default:' gamespeed! (Note: Name can still be anything).");
+			else if (defaultCount > 1)
+				throw new YamlException("Mod defines too many 'default:' gamespeeds!");
```

Comment:
I think this will already throw a key already exists in the dictionary exception.